proposals/20210707-xgboost_evaluator.md

```diff
@@ -0,0 +1,128 @@
+TFX XGBoost Evaluator - Draft Project Proposal
+
+This is the project proposal for the tfx-addons project https://github.com/tensorflow/tfx-addons/issues/13.
+
+# Project Proposal
+
+**Your name:** Daniel Kim
+
+**Your email:** danielk@twitter.com
+
+**Your company/organization:** Twitter
+
+**Project name:** XGBoost Evaluator Component
+# Project Description
+Add support for evaluating XGBoost model in the standard component Evaluator. 
+Add an example pipeline that trains, evaluates and pushes an XGBoost model to CAIP.
+
+## Project Category
+
+Component + Example
+
+## Project Use-Case(s)
+
+This project can be used whenever customers wish to evaluate XGBoost models within a TFX pipeline, in order to obtain the various benefits and functionalities that TFX supports.
+
+## Project Implementation
+
+To make the Evaluator works with XGBoost models, we can customize the Evaluator by providing a Python module with:
+* `custom_eval_shared_model()` to load model artifacts that are not standard TF models
+* `custom_extractors()` to inject a custom prediction extractor. Similar to (tfma.extractors.PredictExtractor)[https://www.tensorflow.org/tfx/model_analysis/api_docs/python/tfma/extractors/PredictExtractor], this extractor uses Beam PTransform to load and extract predictions.
+
+### From Trainer to Evaluator: save and load XGBoost model
+
+Option 1: working with XGBoost library directly
+
+The XGBoost library provides a few different ways to save a model (an xgb.Booster or xgb.sklearn.XGBModel object). Backward compatibility is guaranteed in most cases. Currently, the 2 main supported formats are:
+* XGBoost internal binary format. Note that Auxiliary attributes of the Python Booster object (such as feature_names) will not be loaded when using binary format.
+* JSON: newer format aiming to replace the binary format
```

Comment:
Does this support the "Auxiliary attributes" mentioned above?

Reply:
Yes, the JSON format supports feature_names and other "Auxiliary attributes".